src/rgw/rgw_data_sync.cc

```diff
@@ -1802,10 +1803,11 @@ RGWCoroutine *RGWArchiveDataSyncModule::sync_object(RGWDataSyncEnv *sync_env, RG
     }
   }
 
+  bool stat_follow_olh = true;
```

Comment:
i think this should only be true for the `/* force version if not set */` case

Reply:
Fixed. Thanks!